### PR TITLE
Add test coverage report

### PR DESCRIPTION
**Pull request by mloshakov**
## Summary
Added test coverage plugin

## JIRA Issues
[PILOT-1153](https://indocconsortium.atlassian.net/browse/PILOT-1153?atlOrigin=eyJpIjoiYzRkNjc5OWNiMWNlNDRiNzhkZDY2ZGJlMTBiZGJhZTciLCJwIjoiaiJ9)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [x] No

## Test Directions
None

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -26,8 +29,16 @@ jobs:
...
             poetry-${{ hashFiles('**/poetry.lock') }}
+
       - name: Install dependencies
         run: poetry install
-      - name: Test with pytest
+
+      - name: Test with coverage
         run: |
-          poetry run pytest
+          PYTHONPATH=. poetry run pytest -vvv --cov=common --cov-report term --cov-report=xml
```
Comment (vmoshynskyi) [inline .github/workflows/run_tests.yml:38]: Can you please try to run it without `PYTHONPATH`? I think test will work without it.
```suggestion
          poetry run pytest -vvv --cov=common --cov-report term --cov-report=xml
```

**Review by vmoshynskyi — COMMENTED** [review]
In general everything is fine. Please take a look at my comment.

**Comment by mloshakov** [comment]
Set up pre-commit hooks, two changes that I did manually:

- Updated black to 22.3.0
- Put F401 and F403 to flake8 exceptions, otherwise there are too much files to fix

**Review by vmoshynskyi — CHANGES_REQUESTED** [review]
No, wait. You don't need to update flake8 ignore list or fix everything. 
Please revert everything. Keep only:
1. Changes in `run_tests.yml`
2. Black version update in `.pre-commit-config.yaml`
3. Changes in `README.md`
4. Changes in `pyproject.toml`

**Review by vmoshynskyi — APPROVED** [review]
Thanks!

**Pull request merged**